internal/groups/groups.go

```diff
@@ -0,0 +1,111 @@
+package groups
+
+import (
+	"context"
+)
+
+type Member interface{}
+
+type Metadata map[string]interface{}
+
+type Group struct {
+	ID          string
+	OwnerID     string
+	ParentID    string
+	Name        string
+	Description string
+	Metadata    Metadata
+	Level       int
+	Path        string
```

Comment:
I'm not sure that these belong to the Group struct. If we need info about the position of the Group in the hierarchy, we can expose an API for that. I doubt that we'll need it. Also, please consider limiting the possible depth of this hierarchy as well as the number of direct children to some sane number (10 or 20 should be enough) due to the performance constraints.